frontend/ui/widget/checkbutton.lua

```diff
@@ -109,39 +110,47 @@ function CheckButton:initCheckButton(checked)
 end
 
 function CheckButton:onTapCheckButton()
-    if self.enabled and self.callback then
-        if G_reader_settings:isFalse("flash_ui") then
-            self.callback()
+    if self.enabled then
+        if self.tap_input then
+            self:onInput(self.tap_input)
+        elseif type(self.tap_input_func) == "function" then
+            self:onInput(self.tap_input_func())
         else
-            -- c.f., ui/widget/iconbutton for the canonical documentation about the flash_ui code flow
-
-            local highlight_dimen = self.dimen
-
-            -- Highlight
-            --
-            self[1].invert = true
-            UIManager:widgetInvert(self[1], highlight_dimen.x, highlight_dimen.y, highlight_dimen.w)
-            UIManager:setDirty(nil, "fast", highlight_dimen)
-
-            UIManager:forceRePaint()
-            UIManager:yieldToEPDC()
-
-            -- Unhighlight
-            --
-            self[1].invert = false
-            UIManager:widgetInvert(self[1], highlight_dimen.x, highlight_dimen.y, highlight_dimen.w)
-            UIManager:setDirty(nil, "ui", highlight_dimen)
-
-            -- Callback
-            --
-            self.callback()
-
-            UIManager:forceRePaint()
+            if G_reader_settings:isFalse("flash_ui") then
+                self:toggleCheck()
+                if self.callback then
+                    self.callback()
+                end
+            else
+                -- c.f., ui/widget/iconbutton for the canonical documentation about the flash_ui code flow
+
+                local highlight_dimen = self.dimen
+
+                -- Highlight
+                --
```

Comment:
I was going to ask if there was a way to not have all this indentation changes in @NiLuJe flashing code so we keep seeing its history in git blame.
But all these blank lines and blank comments hurt my eyes :) so, if you're taking ownership of these line with the indentation, you might as well remove these blank comments :)
(Do you really want to see your name associated with this ?!! :):
```
fe10d0bce (NiLuJe         2021-02-20 18:22:48 +0100 116)             -- c.f., ui/widget/iconbutton for the canonical documentation about the flash_ui code flow
fe10d0bce (NiLuJe         2021-02-20 18:22:48 +0100 117)
fe10d0bce (NiLuJe         2021-02-20 18:22:48 +0100 118)             local highlight_dimen = self.dimen
fe10d0bce (NiLuJe         2021-02-20 18:22:48 +0100 119)
fe10d0bce (NiLuJe         2021-02-20 18:22:48 +0100 120)             -- Highlight
fe10d0bce (NiLuJe         2021-02-20 18:22:48 +0100 121)             --
812e59560 (NiLuJe         2019-02-07 01:14:37 +0100 122)             self[1].invert = true
```

Reply:
Indentation restored.